compiler/optimizations/remoteValueForwarding.cpp

```diff
@@ -84,6 +88,83 @@ void remoteValueForwarding() {
   }
 }
 
+static void fixLHSArray(CallExpr* move, std::vector<Symbol*>& todo) {
+  Symbol* LHS = toSymExpr(move->get(1))->symbol();
+  if (LHS->isRef()) {
+    LHS->type = LHS->getValType();
+    LHS->qual = QUAL_VAL;
+    todo.push_back(LHS);
+  }
+}
+
+static void replaceRecordWrappedRefs() {
```

Comment:
I'd really appreciate an overview comment that covers what this function is doing and also why it is OK. E.g.

Replaces references to record-wrapped types with copies of the record-wrapped type. This is valid because array, domain, and distribution records (e.g. _array) are immutable. The difference between ref(_array) and _array is key for the compiler to know when to copy or destroy these objects, but that is handled in callDestructors and so by this point in compilation the distinction is no longer important.